crates/json-abi/src/abi.rs

```diff
@@ -32,22 +51,69 @@ pub struct JsonAbi {
     pub errors: BTreeMap<String, Vec<Error>>,
 }
 
+impl<'a> FromIterator<AbiItem<'a>> for JsonAbi {
+    fn from_iter<T: IntoIterator<Item = AbiItem<'a>>>(iter: T) -> Self {
+        let mut abi = Self::new();
+        for item in iter {
+            let _ = abi.insert_item(item);
+        }
+        abi
+    }
+}
+
 impl JsonAbi {
     /// Creates an empty ABI object.
     #[inline]
     pub fn new() -> Self {
         Self::default()
     }
 
-    /// Parse the ABI json from a `str`. This is a convenience wrapper around
-    /// [`serde_json::from_str`].
+    /// Parse a [Human-Readable ABI] string into a JSON object.
+    ///
+    /// [Human-Readable ABI]: https://docs.ethers.org/v5/api/utils/abi/formats/#abi-formats--human-readable-abi
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use alloy_json_abi::JsonAbi;
+    /// assert_eq!(JsonAbi::parse([])?, JsonAbi::new());
+    ///
+    /// let abi = JsonAbi::parse([
+    ///     "constructor(string symbol, string name)",
+    ///     "function transferFrom(address from, address to, uint value)",
+    ///     "function balanceOf(address owner)(uint balance)",
+    ///     "event Transfer(address indexed from, address indexed to, address value)",
+    ///     "error InsufficientBalance(account owner, uint balance)",
+    ///     "function addPerson(tuple(string, uint16) person)",
+    ///     "function addPeople(tuple(string, uint16)[] person)",
+    ///     "function getPerson(uint id)(tuple(string, uint16))",
+    ///     "event PersonAdded(uint indexed id, tuple(string, uint16) person)",
+    /// ])?;
```

Comment:
I believe we also allowed doing in-line definition of a struct `struct Person { string name; uint16 age }` and then your `addPerson` function would be `addPerson(Person person)` instead of the tuple above. Does this also work if the word `tuple` is missing?

Reply:
1. that's out of scope of this, it parses items one at a time
2. tuple is an optional keyword